### PR TITLE
Feature/search

### PR DESCRIPTION
**Pull request by carinamenke**
This pull request includes a vocab search.

### CODE REVIEW

File: .firebase/hosting.YnVpbGQ.cache [hunk 1]
```diff
@@ -1,35 +1,37 @@
...
+images/spacious.jpg,1584366858223,9658d7fbf2b3702f1290030e5c2d49bb2e40e735f6ff0d53d8b3af434167cabb
+images/plate.jpg,1584366858220,d06fa1c5b5814f854aa19396d7b9074a3cd15c923b3516af2fad651114a51a95
+images/drink.jpg,1584366858217,bd5a81e3392d4fa507d1c297bbc56f139355bfecb04f66032b2a778e0cc73bdf
+images/chair.jpg,1584366858212,7f52e4d6df6877feb2aa246cb46385d16a2ea913e3cfa2091065304123807dcf
+images/sit.jpg,1584366858222,49b50b333d8a19b69f0ef5efe9e8e4010ca057f7514441861483e42e658eea38
+static/js/2.27df42c1.chunk.js,1584366915162,234ab3149baefee309216205a8c93ae75de58672a2761ffcd892b91af659a8de
+images/cosy.jpg,1584366858213,8d159a5efa0f99a4df40e00811fbf05a9d637158b89f0571e03512f4ffc469c6
+images/door.jpg,1584366858216,a889d4fe1585b86949481e2f241f97b255ba09d7412d2f3ef7f62da91410eac8
+images/window.jpg,1584366858225,81bf2cae0ead6cbd842c9cb0f4dbe0821e2df60d1c236d234babdb37973046a8
+static/js/2.27df42c1.chunk.js.map,1584366915163,35fe8e4eb7170427db23cce1ed967e64d5ca4ed3ac548a80b1ee3178bebc9495
```
Comment (albert-schilling) [inline .firebase/hosting.YnVpbGQ.cache]: I am not sure, but I believe it is best to hide this file? Maybe ask Joe about this.

File: README.md [hunk 2]
```diff
@@ -1,79 +1,28 @@
 # vorto app
 
-The vorto app allows you to create a digital stack of vocabulary learning cards. Expand your vocabulary for any foreign language with those words which are relevant for **you**.
+<img src="public/images/vorto_logo.png"  alt="vorto app" width="200" height="100">
+
+The vorto app allows you to create **a digital stack of vocabulary learning cards**. Expand your vocabulary for any foreign language with those words which are relevant for you.
```
Comment (albert-schilling) [inline README.md]: Great that you updated the readme with your logo

File: cypress/integration/searchVocab.spec.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+    cy.get('input[name="q"]').type('The doOr')
+
+    cy.get('.content-title').contains('the door')
+  })
+
+  it('displays no search results when the search input is cleared', () => {
+    cy.get('.closing-icon').click()
+
+    cy.get('main').should('be.empty')
```
Comment (albert-schilling) [inline cypress/integration/searchVocab.spec.js]: Great, you have implemented tests!

File: src/common/GlobalStyles.js [hunk 4]
```diff
@@ -19,6 +19,10 @@ export default createGlobalStyle`
     font-size: 1em;
   }
 
+  button {
+    outline: none;
```
Comment (albert-schilling) [inline src/common/GlobalStyles.js]: Ah, I will probably steal this. It is very annoying to always remove the default button styling.

File: src/common/GlobalStyles.js [hunk 5]
```diff
@@ -19,6 +19,10 @@ export default createGlobalStyle`
...
+  }
+
 
   :root {
```
Comment (albert-schilling) [inline src/common/GlobalStyles.js]: Why don't you put all color defintions in one :root{} declaration?
```
:root {
--backgr...
--back...
...
}
```

File: src/components/AudioButton/AudioButton.js [hunk 6]
```diff
@@ -52,6 +52,6 @@ const AudioButtonStyled = styled.button`
 
   :active {
     background: var(--grey-color-mid2);
-    border: solid var(--grey-color-mid2) 1px;
+    border: solid var(--grey-color-mid2) 1.5px;
```
Comment (albert-schilling) [inline src/components/AudioButton/AudioButton.js]: You only this difference on a retina screen, right?

File: src/components/Navigation/Navigation.js [hunk 7]
```diff
@@ -24,7 +24,7 @@ export default function Navigation() {
 }
 
 const NavigationStyled = styled.nav`
-  grid-row: 3 / 4;
+  grid-row: 4 / 5;
```
Comment (albert-schilling) [inline src/components/Navigation/Navigation.js]: Oh, I do not know this. What does this say?

File: src/components/PrimaryButton/PrimaryButton.js [hunk 8]
```diff
@@ -21,8 +21,9 @@ const PrimaryButtonStyled = styled.button`
   color: var(--text-color-white);
   border: none;
   border-radius: 4px;
-  margin: 20px 15px;
+  margin: 15px 15px;
   padding: 7px;
+  min-height: 35px;
```
Comment (albert-schilling) [inline src/components/PrimaryButton/PrimaryButton.js]: Why min-height and not just height?

File: src/components/PrimaryButton/PrimaryButton.js [hunk 9]
```diff
@@ -21,8 +21,9 @@ const PrimaryButtonStyled = styled.button`
   color: var(--text-color-white);
   border: none;
   border-radius: 4px;
-  margin: 20px 15px;
+  margin: 15px 15px;
```
Comment (albert-schilling) [inline src/components/PrimaryButton/PrimaryButton.js]: `margin: 15px;` would say the same.

File: src/components/Search/Search.js [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+        {searchInput && (
+          <ClearButtonStyled onClick={handleReset}>
+            <FiX className="closing-icon" />
+          </ClearButtonStyled>
+        )}
+      </SearchFieldStyled>
+      {searchInput && <span>{searchResult.length} vocabularies found:</span>}
+    </SearchStyled>
+  )
+}
```
Comment (albert-schilling) [inline src/components/Search/Search.js]: The search comp. looks really nice and short. Well done!

File: src/components/Search/Search.stories.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+    renderSearch => (
+      <div style={{ padding: 20, width: 400 }}>{renderSearch()}</div>
+    ),
+  ],
+}
+
+export const Default_Search = () => (
+  <Search
+    handleSearch={action('Searches for search term')}
+    handleReset={action('Clears search field')}
```
Comment (albert-schilling) [inline src/components/Search/Search.stories.js]: Great use of Storybook and actions, so one understands from the styleguide what the comp should do functionwise.

File: src/pages/FormPage.js [hunk 12]
```diff
@@ -8,6 +9,10 @@ import InputSelect from '../components/InputSelect/InputSelect'
 import InputText from '../components/InputText/InputText'
 import PrimaryButton from '../components/PrimaryButton/PrimaryButton'
```
Comment (albert-schilling) [inline src/pages/FormPage.js]: Nice use of smaller components for specific input types.

File: src/pages/FormPage.js [hunk 13]
```diff
@@ -1,3 +1,4 @@
+import PropTypes from 'prop-types'
```
Comment (albert-schilling) [inline src/pages/FormPage.js]: Nice that you consistently use prop-types!

File: src/pages/SearchPage.js [hunk 14]
```diff
@@ -1,5 +1,58 @@
...
+      filteredVocabs = allVocabs.filter(
+        vocab =>
+          vocab.wordTitle.includes(searchInput.toLowerCase()) ||
+          vocab.translation.includes(searchInput.toLowerCase())
+      )
+      setSearchResult(filteredVocabs)
+    } else {
+      setSearchResult([])
+    }
+  }, [searchInput, vocabs])
```
Comment (albert-schilling) [inline src/pages/SearchPage.js]: Nice and logical use of useEffect with the dependencies. Search changes: restart filter and setState. Words changed: (restart filter) and setState. Great!

File: .firebase/hosting.YnVpbGQ.cache (hunk 1 above)
Comment (carinamenke) [reply .firebase/hosting.YnVpbGQ.cache]: Yes, seems like adding it to .gitignore is not enough/ didn't work? I will check :)

File: src/common/GlobalStyles.js (hunk 5 above)
Comment (carinamenke) [reply src/common/GlobalStyles.js]: ahah, good point! You're right, this is super redundant, I will change it. Thanks :)

File: src/components/AudioButton/AudioButton.js (hunk 6 above)
Comment (carinamenke) [reply src/components/AudioButton/AudioButton.js]: Probably, I did see a difference on my screen :)

File: src/components/PrimaryButton/PrimaryButton.js (hunk 9 above)
Comment (carinamenke) [reply src/components/PrimaryButton/PrimaryButton.js]: oh, I overlooked this, thanks for the hint :)

File: src/components/Navigation/Navigation.js (hunk 7 above)
Comment (carinamenke) [reply src/components/Navigation/Navigation.js]: I have a grid in the App.js with 4 rows. In the other components I then indicate via grid-row, which rows exactly they should fill. In this case it's from line 4 to 5, which is the 4th row at the bottom of the screen. https://css-tricks.com/snippets/css/complete-guide-grid/#prop-grid-column-row

File: src/components/PrimaryButton/PrimaryButton.js (hunk 8 above)
Comment (carinamenke) [reply src/components/PrimaryButton/PrimaryButton.js]: Not sure anymore why I did this, height is sufficient indeed :)

File: .firebase/hosting.YnVpbGQ.cache (hunk 1 above)
Comment (albert-schilling) [reply .firebase/hosting.YnVpbGQ.cache]: Might be commited before you introduced it to the gitignore file?

File: src/components/Navigation/Navigation.js (hunk 7 above)
Comment (albert-schilling) [reply src/components/Navigation/Navigation.js]: Cool, thx for the explanation!

File: .firebase/hosting.YnVpbGQ.cache (hunk 1 above)
Comment (carinamenke) [reply .firebase/hosting.YnVpbGQ.cache]: Yeah, I think that's probably the reason. I deleted the firebase files from github now

**Pull request merged**